src/rules/require-baseline.js

```diff
@@ -339,6 +339,21 @@ class SupportsRules {
 	}
 }
 
+/**
+ * Decodes and extracts the status and year from an encoded feature status.
+ * @param {string} encodedStatus The encoded feature status and year.
+ * @returns {Object} An object containing the feature status and year.
+ */
+function decodeBaselineStatus(encodedStatus) {
+	const match = encodedStatus.match(
+		/^(?<status>\d+)(?: \((?<year>\d{4})\))?$/u,
+	);
+	const status = Number(match.groups.status);
+	const year = Number(match.groups.year);
```

Comment:
I think we can move these lines directly into the `isSupported()` method. That way, we can save the object created as the return value and we can also avoid running this same logic twice for each feature. What do you think?

Reply:
The logic should only run one time per feature, thanks to the early return.

Something like might be clearer:

```js
isSupported(feature) {
	const { status, year } = decodeBaselineStatus(feature);
	
	if (this.#baselineYear) {
		return year <= this.#baselineYear;
	}

	return status >= this.#baselineStatus;
}
```

Either way, I'm happy to merge this with `isSupported` if you still prefer.

Reply:
Ok, updated